toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -779,6 +791,7 @@ func extractKernelCmdlineFromGrub(bootPartition diskutils.PartitionInfo, bootDir
 	grubCfgPath := filepath.Join(tmpDirBoot, bootDirPath, DefaultGrubCfgPath)
 	kernelToArgs, err := extractKernelCmdlineFromGrubFile(grubCfgPath)
 	if err != nil {
+		_ = bootPartitionMount.CleanClose() // Synchronous cleanup on error
```

Comment:
`CleanClose` can fail. And when it does, the state of the disk is unknown.

Please change the code to return a `bool` when the grub config file is missing.